src/main/java/net/dries007/tfc/compat/jade/common/BlockEntityTooltips.java

```diff
@@ -0,0 +1,332 @@
+            LampFuel fuel = lamp.getFuel();
+            if (fuel != null)
+            {
+                tooltip.accept(Helpers.translatable("tfc.jade.burn_rate", String.valueOf(fuel.getBurnRate())));
+                if (state.getValue(LampBlock.LIT))
+                {
+                    if (fuel.getBurnRate() == -1)
+                    {
+                        tooltip.accept(Helpers.translatable("tfc.jade.burn_forever"));
+                    }
+                    else
+                    {
+                        lamp.getCapability(Capabilities.FLUID).ifPresent(cap -> {
+                            final int fluid = cap.getFluidInTank(0).getAmount();
+                            if (fluid > 0)
+                            {
+                                // ticks / mB * mB = ticks
+                                tooltip.accept(Helpers.translatable("tfc.jade.hours_remaining", String.valueOf(Math.round((float) fluid * fuel.getBurnRate() / ICalendar.TICKS_IN_HOUR))));
+                            }
+                        });
+                    }
+
+                }
+
+            }
+        }
+    };
+
+    public static final BlockEntityTooltip NEST_BOX = (level, state, entity, tooltip) -> {
+        if (entity instanceof NestBoxBlockEntity nest)
+        {
+            final List<Component> text = new ArrayList<>();
+            nest.getCapability(Capabilities.ITEM).ifPresent(inventory -> {
+                for (int i = 0; i < inventory.getSlots(); i++)
+                {
+                    ItemStack stack = inventory.getStackInSlot(i);
+                    stack.getCapability(EggCapability.CAPABILITY).filter(IEgg::isFertilized).ifPresent(egg -> {
+                        text.add(stack.getHoverName());
+                        egg.addTooltipInfo(text);
+                    });
+                }
+            });
+            text.forEach(tooltip);
+        }
+    };
+
+    public static final BlockEntityTooltip PIT_KILN_INTERNAL = (level, state, entity, tooltip) -> pitKiln(level, entity, tooltip, 0);
+    public static final BlockEntityTooltip PIT_KILN_ABOVE = (level, state, entity, tooltip) -> pitKiln(level, entity, tooltip, -1);
+
+    public static final BlockEntityTooltip POWDER_KEG = (level, state, entity, tooltip) -> {
+        if (entity instanceof PowderkegBlockEntity keg)
+        {
+            tooltip.accept(Helpers.translatable("tfc.jade.explosion_strength").append(Helpers.translatable(String.valueOf(PowderkegBlockEntity.getStrength(keg)))));
+        }
+    };
+
+    public static final BlockEntityTooltip SAPLING = (level, state, entity, tooltip) -> {
+        if (entity instanceof TickCounterBlockEntity counter && state.getBlock() instanceof TFCSaplingBlock sapling)
+        {
+            final String growth = Helpers.formatPercentage(Math.min(0.99f, (float) counter.getTicksSinceUpdate() / (sapling.getDaysToGrow() * ICalendar.TICKS_IN_DAY)) * 100);
```

Comment:
Again with the percentage, I think just some sort of "Growing" / "Not Growing" (with reason) is more helpful than a percentage, and the percent growth is misleading.